avocado_vt/vt_agent/src/avocado_vt/agent/managers/resource_backing_manager.py

```
@@ -0,0 +1,242 @@
+        # the backing manager has to restore the configurations. Abort the
+        # process whenever there is a loading failure
+        try:
+            if os.path.isfile(BACKING_MGR_ENV_FILENAME):
+                self._load()
+        except Exception as e:
+            LOG.error("Failed to load the backing manager env file: %s", str(e))
+            raise
+
+    def _load(self):
+        with open(BACKING_MGR_ENV_FILENAME, "rb") as f:
+            self._dump_data = pickle.load(f)
```

Ruff (0.13.1)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)